core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -539,24 +540,68 @@ class CompactEntityQuery(driverComponent: DriverComponent) extends RawSqlQuery w
         reduceSqlActionsWithDelim(replaceParamsSqls.toSeq, sql","),
         sql""") where er.workspace_id = $workspaceId and er.to_entity_type = $oldType"""
       )
-  }
-    
+    }
+
     // Execute all updates in same transaction
     for {
       paths <- getReferencePathsInAttributesSql.as[String]
-      _ <- if(paths.isEmpty) {
-        DBIO.successful(0)
-      } else {
-        DBIO.seq(
-          updateReferencesInAttributesSql(paths).asUpdate,
-          updateToReferencesSql.asUpdate
-        )
-      }
+      _ <-
+        if (paths.isEmpty) {
+          DBIO.successful(0)
+        } else {
+          DBIO.seq(
+            updateReferencesInAttributesSql(paths).asUpdate,
+            updateToReferencesSql.asUpdate
+          )
+        }
       _ <- updateFromReferencesSql.asUpdate
       entityRowsUpdated <- updateEntityTypeSql.asUpdate
     } yield entityRowsUpdated
   }
-  
+
+  /**
+    * Renames a single attribute across all entities of the given type and workspace.
+    * This method assumes the old and new attribute names have already been validated!
+    *
+    * `Using where. Index used: idx_entity_type_name`
+    */
+  def renameAttribute(workspaceId: UUID,
+                      entityType: String,
+                      oldAttributeName: AttributeName,
+                      renameRequest: AttributeRename
+  ): ReadWriteAction[Int] =
+    // rename is implemented as JSON_REMOVE(JSON_SET(JSON_EXTRACT))
+    // JSON_EXTRACT gets the value of the old attribute
+    // JSON_SET creates the new attribute with that value
+    // JSON_REMOVE deletes the old attribute
+    sql"""update ENTITY
+          set record_version = record_version + 1,
+          attributes = JSON_REMOVE(
+                         JSON_SET(
+                           attributes,
+                           '#${slickAttributePath(renameRequest.newAttributeName)}',
```

Comment:
this is not sql injection-safe, get rid of the #

Reply:
fixed in [bcea547](https://github.com/broadinstitute/rawls/pull/3317/commits/bcea547288b963cc10cf291b4fd77e7b0380f5e7)